frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js

```diff
@@ -150,3 +150,30 @@ export function visitQuestion(id) {
 
   cy.wait("@" + alias);
 }
+
+/**
+ * Visit a dashboard and wait for its query to load.
+ *
+ * NOTE: Avoid using this helper if you need to explicitly wait for
+ * and assert on the individual dashcard queries.
+ *
+ * @param {number} id
+ */
+export function visitDashboard(id) {
+  cy.intercept("GET", `/api/dashboard/${id}`).as("getDashboard");
+  // The very last request when visiting dashboard always checks the collection it is in.
+  // That is - IF user has the permission to view that dashboard!
+  cy.intercept("GET", `/api/collection/*`).as("getParentCollection");
+
+  cy.visit(`/dashboard/${id}`);
+
+  // If users doesn't have permissions to even view the dashboard,
+  // the last request for them would be `getDashboard`.
+  cy.wait("@getDashboard").then(({ response: { statusCode } }) => {
```

Comment:
For example:
![image](https://user-images.githubusercontent.com/31325167/159998947-2c3bfd62-af23-4bb0-9d84-971aea27edad.png)

Prior to this adjustment, [this test](https://github.com/metabase/metabase/blob/f3432569e8665044577c62e2ddf95f371031987b/frontend/test/metabase/scenarios/dashboard/permissions.cy.spec.js#L115) failed
https://app.circleci.com/pipelines/github/metabase/metabase/31571/workflows/be70cbef-a36e-49db-8477-f69c4ccbb2fd/jobs/1601335/artifacts